xdsl/dialects/mpi.py

```diff
@@ -338,9 +372,58 @@ class Init(MPIBaseOp):
 
 @irdl_op_definition
 class Finalize(MPIBaseOp):
+    """
+    This represents an MPI_Finalize call with both args being nullptr
+    """
     name = "mpi.finalize"
 
 
+@irdl_op_definition
+class UnwrapMemrefOp(MPIBaseOp):
+    """
+    This Op can be used as a helper to get memrefs into MPI calls.
+
+    It takes any MemRef as input, and returns an llvm.ptr, element count and MPI_Datatype.
+    """
+    name = "mpi.unwrap_memref"
+
+    ref: Annotated[Operand, MemRefType[AnyNumericType]]
+
+    ptr: Annotated[OpResult, llvm.LLVMPointerType]
+    len: Annotated[OpResult, i32]
+    typ: Annotated[OpResult, DataType]
+
+    @staticmethod
+    def get(ref: SSAValue | Operation):
+        ssa_val = SSAValue.get(ref)
+        assert isinstance(ssa_val.typ, MemRefType)
+        elem_typ = cast(MemRefType[AnyNumericType], ssa_val.typ).element_type
+
+        return UnwrapMemrefOp.build(operands=[ref],
+                                    result_types=[
+                                        llvm.LLVMPointerType.typed(elem_typ),
+                                        i32,
+                                        DataType()
+                                    ])
+
+
+@irdl_op_definition
+class GetDtypeOp(MPIBaseOp):
```

Comment:
Feels like this also needs an Operand. Or maybe we actually need an actual cast operation in a more general dialect?

Reply:
I feel like this op would be clearer to me if you could also provide a small IR snippet in the docstring.

Reply:
Will do!

Reply:
Done!

Reply:
Ah okay, I get know what this does :+1: